src/Microsoft.AspNet.OData/Routing/ODataActionSelector.cs

```diff
@@ -106,5 +112,80 @@ private static HttpResponseMessage CreateErrorResponse(HttpRequestMessage reques
 
             return request.CreateErrorResponse(statusCode, error);
         }
+
+        private static bool ActionParametersMatchRequest(HttpActionDescriptor action, HttpControllerContext context)
+        {
+            var parameters = action.GetParameters();
+            var routeData = context.RouteData;
+            var conventionsStore = context.Request.ODataProperties().RoutingConventionsStore;
+            var matchedBody = false;
+            var route = routeData.Route as ODataRoute;
+            var routePrefix = route?.RoutePrefix;
+            var availableKeys = routeData.Values.Keys
+                .Where(k => routePrefix != "{" + k + "}"
+                    && k != ODataRouteConstants.Action
+                    && k != ODataRouteConstants.Controller
+                    && k != ODataRouteConstants.ODataPath)
+                .Select(k => k.ToUpperInvariant())
+                .ToList();
+
+            if (parameters.Count == 0 && availableKeys.Count > 0)
```

Comment:
>availableKeys.Count > 0 [](start = 41, length = 23)

more efficient to do availableKeys.Any(); you don't care about the actual count, so no need to count them. you just want to know if there are any, and any can return after it finds the first one.